plugins/octant/cmd/traceflow-plugin/main.go

```diff
@@ -0,0 +1,409 @@
+	return navigation.Navigation{
+		Title:    "Trace Flow",
+		Path:     request.GeneratePath("components"),
+		IconName: "cloud",
+	}, nil
+}
+
+func (a *traceflowPlugin) regExpMatch(pattern, str string) bool {
+	match, err := regexp.MatchString(pattern, str)
+	if err != nil {
+		log.Printf("unable to judge srcPod string pattern: %s", err)
+		return false
+	}
+	if !match {
+		log.Printf("string %s and regExp pattern %s does not match", str, pattern)
+		return false
+	}
+	return true
+}
+
+func (a *traceflowPlugin) actionHandler(request *service.ActionRequest) error {
+	actionName, err := request.Payload.String("action")
+	if err != nil {
+		log.Printf("unable to get input at string: %s", err)
+		return nil
+	}
+
+	switch actionName {
+	case addTfAction:
+		srcNamespace, err := request.Payload.String(srcNamespaceCol)
+		if err != nil {
+			log.Printf("unable to get srcNamespace at string : %s", err)
+		}
+		if match := a.regExpMatch(namespaceStrPattern, srcNamespace); !match {
+			return nil
+		}
+
+		srcPod, err := request.Payload.String(srcPodCol)
+		if err != nil {
+			log.Printf("unable to get srcPod at string : %s", err)
+		}
+		if match := a.regExpMatch(podStrPattern, srcPod); !match {
+			return nil
+		}
+
+		srcPort, err := request.Payload.String(srcPortCol)
+		if err != nil {
+			log.Printf("unable to get srcPort at string : %s", err)
+		}
+
+		dstNamespace, err := request.Payload.String(dstNamespaceCol)
+		if err != nil {
+			log.Printf("unable to get dstNamespace at string : %s", err)
+		}
+		if match := a.regExpMatch(namespaceStrPattern, dstNamespace); !match {
+			return nil
+		}
+
+		dstPod, err := request.Payload.String(dstPodCol)
+		if err != nil {
+			log.Printf("unable to get dstPod at string : %s", err)
+		}
+		if match := a.regExpMatch(podStrPattern, dstPod); !match {
+			return nil
+		}
+
+		// TODO: "Destination is a service" is not supported now.
+		/*	dstService, err := request.Payload.String(dstSvcCol)
+			if err != nil {
+				log.Printf("unable to get dstService at string : %s", err)
+			}*/
+
+		dstPort, err := request.Payload.String(dstPortCol)
+		if err != nil {
+			log.Printf("unable to get dstPort at string : %s", err)
+		}
+
+		protocol, err := request.Payload.String(protocolCol)
+		if err != nil {
+			log.Printf("unable to get dstPod at string : %s", err)
+		}
+		protocol = strings.ToUpper(protocol)
+
+		// Judge whether the name of trace flow is duplicated.
+		// If it is, then the user creates more than one traceflows in one second, which is not allowed.
+		tfName := srcPod + "-" + dstPod + "-" + time.Now().Format(TIME_FORMAT_YYYYMMDD_HHMMSS)
+		ctx := context.Background()
+		tfOld, err := a.client.OpsV1alpha1().Traceflows().Get(ctx, tfName, metav1.GetOptions{})
+		if err != nil {
+			log.Printf("unable to get traceflow \"%s\", detailed error: %s", tfName, err)
+		}
+		if tfOld.Name == tfName {
+			log.Printf("duplicate traceflow \"%s\": %+v. ", tfName, tfOld)
```

Comment:
Better to log information about the one second limit too.

Reply:
Fixed.